Product/Production/Services/AuditRepositoryCore/src/main/java/gov/hhs/fha/nhinc/auditrepository/hibernate/AuditRepositoryDAO.java

```diff
@@ -161,4 +167,189 @@ public List queryAuditRepositoryOnCriteria(String eUserId, String ePatientId, Da
+
+            if (outcome != null && outcome >= 0) {
+                queryCriteria.add(Restrictions.eq("outcome", outcome));
+            }
+
+            if (NullChecker.isNotNullish(eventTypeList)) {
+                queryCriteria.add(Restrictions.in("eventType", eventTypeList));
+            }
+
+            if (NullChecker.isNotNullish(userId)) {
+                queryCriteria.add(Restrictions.eq("userId", userId));
+            }
+
+            if (NullChecker.isNotNullish(remoteHcidList)) {
+                queryCriteria.add(Restrictions.in("remoteHcid", remoteHcidList));
+            }
+
+            if (NullChecker.isNotNullish(startDate) && NullChecker.isNotNullish(endDate)) {
+                SimpleDateFormat sdf = new SimpleDateFormat("MM/dd/yyyy");
+
+                queryCriteria.add(Restrictions.between("eventTimestamp", new Date(sdf.parse(startDate).getTime()),
+                    new Date(sdf.parse(endDate).getTime())));
+
+            } else if (NullChecker.isNotNullish(startDate) && NullChecker.isNullish(endDate)) {
+                SimpleDateFormat sdf = new SimpleDateFormat("MM/dd/yyyy");
+
+                queryCriteria.add(Restrictions.ge("eventTimestamp", new Date(sdf.parse(startDate).getTime())));
+
+            } else if (NullChecker.isNullish(startDate) && NullChecker.isNotNullish(endDate)) {
+                SimpleDateFormat sdf = new SimpleDateFormat("MM/dd/yyyy");
+
+                queryCriteria.add(Restrictions.le("eventTimestamp", new Date(sdf.parse(endDate).getTime())));
+
+            }
+
+            queryCriteria.setProjection(Projections.projectionList()
+                .add(Projections.property("userId"))
+                .add(Projections.property("eventType"))
+                .add(Projections.property("eventId"))
+                .add(Projections.property("outcome"))
+                .add(Projections.property("eventTimestamp"))
+                .add(Projections.property("remoteHcid"))
+                .add(Projections.property("relatesTo"))
+                .add(Projections.property("direction"))
+                .add(Projections.property("id"))
+                .add(Projections.property("messageId")));
+
+            // if no criteria is passed then it will search full database with above mentioned columns in the result
+            queryList = queryCriteria.list();
+
+        } catch (HibernateException | ParseException e) {
+            LOG.error("Exception in AuditLog.get() occurred due to :" + e.getLocalizedMessage(), e);
+        } finally {
+            // Actual contact insertion will happen at this step
+            if (session != null) {
+                session.flush();
+                session.close();
+            }
+        }
+        return queryList;
+    }
+
+    /**
+     * This method does a query to database to get the Audit Log Blob Messages based on the auditId
+     *
+     * @param auditId
+     * @return List
+     */
+    public List queryAuditViewerByAuditId(String auditId) {
```

Comment:
Why is this method expect a list.  If you provide an id, you should get one and only one result back.